### PR TITLE
Update the OCS Share API Get Shares From Specific File or Folder docs

### LINKED ISSUE #1193: [dev] [OCS] API for pending shares in 10.0.9+
## What Needs to be Documented?

10.0.9 introduced pending shares. This also needs to be added to the developer docs. I'd assume it looks similar to pending [federated shares](https://doc.owncloud.com/server/developer_manual/core/ocs-share-api.html#federated-cloud-shares)

## Where Does This Need To Be Documented?

https://doc.owncloud.com/server/developer_manual/core/ocs-share-api.html

## Why Should This Change Be Made? (Optional)

Information is needed for implementation in ownCloud clients.

## What Type Of Content Change Is This? (Optional)
- [x] New Content Addition
- [ ] Old Content Deprecation
- [ ] Existing Content Simplification
- [ ] Bug Fix to Existing Content

## Which Manual Does This Relate To? (Optional)
- [ ] Admin Manual
- [x] Developer Manual
- [ ] User Manual
- [ ] Android
- [ ] iOS
- [ ] Branded Clients
- [ ] Desktop Client
- [ ] Other

/cc @DeepDiver1975

### PR DESCRIPTION
**Pull request by settermjd**
This change updates the request attributes table for the OCS Share API get shares from specific file or folder documentation. Specifically, it adds the shared_with_me, format, and state attributes; two of which are required for the pending shares API in oC 10.0.9+. This change, therefore, relates to and fixes #1193 and #286.

### CODE REVIEW

File: modules/developer_manual/examples/core/scripts/ruby/decline-pending-share.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require 'net/http'
+require 'uri'
+
+base_uri = '{oc-examples-server-url}/ocs/v1.php/apps/files_sharing/api/v1/'
+uri = URI("#{base_uri}/shares/115468")
```
Comment (jvillafanez) [inline modules/developer_manual/examples/core/scripts/ruby/decline-pending-share.rb]: the uri doesn't seem right

File: modules/developer_manual/pages/core/apis/ocs-share-api.adoc [hunk 2]
```diff
@@ -289,6 +243,243 @@ include::{examplesdir}core/scripts/responses/shares/get-share-info-success.xml[]
 
 For details about the elements in the XML response payload please refer to the Response Attributes section of xref:response-attributes-1[the Create a New Share section] below.
 
+=== Accept a Pending Share
+
+* Syntax: `/shares/<share_id>`
```
Comment (jvillafanez) [inline modules/developer_manual/pages/core/apis/ocs-share-api.adoc]: this doesn't seem right

File: modules/developer_manual/pages/core/apis/ocs-share-api.adoc [hunk 3]
```diff
@@ -289,6 +243,243 @@ include::{examplesdir}core/scripts/responses/shares/get-share-info-success.xml[]
...
+[source,go,subs="attributes+"]
+----
+include::{examplesdir}core/scripts/go/accept-pending-share.go[]
+----
+--
+====
+
+=== Decline a Pending Share
+
+* Syntax: `/shares/<share_id>`
```
Comment (jvillafanez) [inline modules/developer_manual/pages/core/apis/ocs-share-api.adoc]: this doesn't seem right

File: modules/developer_manual/examples/core/scripts/ruby/decline-pending-share.rb (hunk 1 above)
Comment (settermjd) [reply modules/developer_manual/examples/core/scripts/ruby/decline-pending-share.rb]: I thought you were wrong at first, as it took me a bit to see the missing path segment (`/pending`). Thanks for pointing that out.

**Comment by michaelstingl** [comment]
@settermjd problem with the `package-lock.json`? This is what I get:

```
$ yarn install
yarn install v1.16.0
warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
✨  Done in 1.83s.
```

Steps to reproduce:

```
  508  rm -rf docs
  509  git clone git@github.com:owncloud/docs.git
  510  cd docs/
  511  git checkout update-ocs-share-api-get-shares-from-a-specific-file-or-folder
  512  git pull
  513  yarn install
```

**Comment by michaelstingl** [comment]
@settermjd I've seen you modified [`examples/core/scripts/curl/list-all-shares.sh`](https://github.com/owncloud/docs/blob/695a58f413deb6311e43f630c6d83e1587b78246/modules/developer_manual/examples/core/scripts/curl/list-all-shares.sh), but it's not in use. Could you check?

**Comment by settermjd** [comment]
@michaelstingl, thanks for [pointing out](https://github.com/owncloud/docs/pull/1570#issuecomment-517418456) that two files are, effectively, orphaned. I'll remove them in a separate PR.

**Comment by settermjd** [comment]
@michaelstingl, thanks for the inspiration. I've created a small shell script in #1604 to check for orphaned example scripts.

**Comment by settermjd** [comment]
@michaelstingl, package-lock.json is slated for removal in #1605.

**Comment by settermjd** [comment]
@jvillafanez @michaelstingl can I get a further review?

File: modules/developer_manual/pages/core/apis/ocs-share-api.adoc [hunk 4]
```diff
@@ -1,42 +1,109 @@
...
+|boolean
+|returns not only the shares shared with the current user but all shares.
+
+|`shared_with_me`
+|string
+|limits the returned shares to only those shared with the authenticating user.
+
+|`state`
+|string
+a|limits the returned shares to only those with the specified state.
```
Comment (jvillafanez) [inline modules/developer_manual/pages/core/apis/ocs-share-api.adoc]: that "a" seems a typo

File: modules/developer_manual/examples/core/scripts/ruby/decline-pending-share.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+require 'net/http'
+require 'uri'
+
+base_uri = '{oc-examples-server-url}/ocs/v1.php/apps/files_sharing/api/v1/'
+uri = URI("#{base_uri}/shares/pending/115468")
```
Comment (jvillafanez) [inline modules/developer_manual/examples/core/scripts/ruby/decline-pending-share.rb]: kind of weird that the rest of the example has a "/pending/1" except this one.

**Comment by jvillafanez** [comment]
content-wise it looks fine, just minor optional touches.

File: modules/developer_manual/examples/core/scripts/ruby/decline-pending-share.rb (hunk 5 above)
Comment (settermjd) [reply modules/developer_manual/examples/core/scripts/ruby/decline-pending-share.rb]: Thanks for spotting this.

**Pull request merged**